tests/test_validation.py

```diff
@@ -1,514 +1,614 @@
-                        rule_args = f" MockComponent.{node_label} Patient.PatientID"
-                    elif second_rule.startswith("inRange"):
-                        rule_args = " 1 1000 warning"
-                    elif second_rule.startswith("regex"):
-                        rule_args = " search [a-f]"
-                    else:
-                        rule_args = ""
-                    # Edit the node info -in place-
-                    node_info["validationRules"].append(second_rule + rule_args)
-                    break
-
-        # Update the manifest to only contain the Component and attribute column where the rule was changed.
-        manifest = manifest[["Component", attribute]]
-
-        data_model_js = DataModelJSONSchema(
-            jsonld_path=helpers.get_data_path("example.model.jsonld"), graph=dmge.graph
-        )
-        json_schema = data_model_js.get_json_validation_schema(
-            source_node=rootNode, schema_name=rootNode + "_validation"
-        )
-
-        validateManifest = ValidateManifest(
-            errors=[],
-            manifest=manifest,
-            manifestPath=manifestPath,
-            dmge=dmge,
-            jsonSchema=json_schema,
-        )
-
-        # perform validation with no exceptions raised
-        _, errors, warnings = validateManifest.validate_manifest_rules(
-            manifest=manifest,
-            dmge=dmge,
-            restrict_rules=False,
-            project_scope=None,
-        )
+import os
+import logging
+import re
+import networkx as nx
+import jsonschema
+import pytest
+from pathlib import Path
+import itertools
+from pandas import NA
+
+from schematic.models.validate_attribute import ValidateAttribute, GenerateError
+from schematic.models.validate_manifest import ValidateManifest
+from schematic.models.metadata import MetadataModel
+from schematic.store.synapse import SynapseStorage
+
+from schematic.schemas.data_model_parser import DataModelParser
+from schematic.schemas.data_model_graph import DataModelGraph, DataModelGraphExplorer
+from schematic.schemas.data_model_json_schema import DataModelJSONSchema
+
+from schematic.utils.validate_rules_utils import validation_rule_info
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(name="dmge")
+def DMGE(helpers):
+    dmge = helpers.get_data_model_graph_explorer(path="example.model.jsonld")
+    yield dmge
+
+@pytest.fixture(name="missing_dmge")
+def missingDMGE(helpers):
+    missing_dmge = helpers.get_data_model_graph_explorer(path="example.missing.value.model.jsonld")
+    yield missing_dmge
+
+@pytest.fixture
+def metadataModel(helpers):
+    metadataModel = MetadataModel(
+        inputMModelLocation=helpers.get_data_path("example.model.jsonld"),
+        inputMModelLocationType="local",
+        data_model_labels="class_label",
+    )
+
+    yield metadataModel
+
+@pytest.fixture
+def missingMetadataModel(helpers):

```

Comment:
same comment as above but in this case something like: `missing_value_model_MetadataModel`